etna/transforms/math/binary_operator.py

```diff
@@ -0,0 +1,181 @@
+from enum import Enum
+from typing import List
+from typing import Optional
+
+import pandas as pd
+
+from etna.datasets import TSDataset
+from etna.transforms.base import ReversibleTransform
+
+
+class BinaryOperator(str, Enum):
+    """Enum for mathematical operators from pandas."""
+
+    add = "+"
+    sub = "-"
+    mul = "*"
+    div = "/"
+    floordiv = "/="
+    mod = "%"
+    pow = "**"
+
+    eq = "=="
+    ne = "!="
+    le = "<="
+    lt = "<"
+    ge = ">="
+    gt = ">"
+
+    @classmethod
+    def _missing_(cls, value):
+        raise ValueError(f"Supported operands: {', '.join([repr(m.value) for m in cls])}.")
+
+    def perform(self, df: pd.DataFrame, left_operand: str, right_operand: str, out_column: str) -> pd.DataFrame:
+        """Perform binary operation on passed dataframe.
+        - If during the operation a division by zero of a positive number occurs, writes +inf to this cell of the column, if negative - -inf.
+        - In the case of raising a negative number to a non-integer power, writes nan to this cell of the column.
+
+        Parameters
+        ----------
+        df:
+            Source Dataframe
+        left_operand:
+            Name of the left column
+        right_operand:
+            Name of the right column
+        out_column:
+            Resulting column name, which contains the result of the operation operand(left, right)
+        """
+        pandas_operator = getattr(pd.DataFrame, self.name)
+        df_left = df.loc[:, pd.IndexSlice[:, left_operand]].rename(columns={left_operand: out_column}, level="feature")
+        df_right = df.loc[:, pd.IndexSlice[:, right_operand]].rename(
+            columns={right_operand: out_column}, level="feature"
+        )
+        return pandas_operator(df_left, df_right)
+
+
+class BinaryOperationTransform(ReversibleTransform):
+    """Perform binary operation on the columns of dataset."""
+
+    def __init__(self, left_column: str, right_column: str, operator: str, out_column: Optional[str] = None):
```

Comment:
We should probably clarify for which operations we have inverse transforms and for which we doesn't have.